File: python/leetcode/156.py
class Solution:
    def upsideDownBinaryTree(self, root):
        if not root or not root.left:
            return root
        left = root.left
        right = root.right
        nroot = self.upsideDownBinaryTree(left)
        left.right = root
        left.left = right
        root.left = None
        root.right = None
        return nroot

File: cpp/leetcode/156.cpp
class Solution {
public:
    TreeNode *upsideDownBinaryTree(TreeNode *root) {
        if (!root || !root->left) {
            return root;
        }
        TreeNode *left = root->left;
        TreeNode *right = root->right;
        TreeNode *nroot = upsideDownBinaryTree(left);
        left->right = root;
        left->left = right;
        root->left = NULL;
        root->right = NULL;
        return nroot;
    }
};